apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -454,22 +469,26 @@ private AbstractTracingSpan pop() {
      *
      * @param span
      */
-    private AbstractTracingSpan push(AbstractTracingSpan span) {
+    private AbstractSpan push(AbstractSpan span) {
         activeSpanStack.addLast(span);
         return span;
     }
 
     /**
      * @return the top element of 'ActiveSpanStack' only.
      */
-    private AbstractTracingSpan peek() {
+    private AbstractSpan peek() {
         if (activeSpanStack.isEmpty()) {
             return null;
         }
         return activeSpanStack.getLast();
     }
 
-    private AbstractTracingSpan first() {
+    private AbstractSpan first() {
         return activeSpanStack.getFirst();
     }
+
+    private boolean isLimitMechanismWorking() {
+        return spanIdGenerator > Config.Agent.SPAN_LIMIT_PER_SEGMENT;
```

Comment:
I think this express should be `>=`, because the `spanIdGenerator` value is start from 0.

Reply:
@acurtain You can change that through a pull request. If this is a bug, we have nothing to discuss, just need a fix. Right?